__tests__/app.test.js

```diff
@@ -149,70 +149,93 @@ describe("/api/articles/:article_id", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          if (articles.article_id === 1) {
+            expect(articles.comment_count).toBe(11);
+          }
+          if (articles.article_id === 9) {
+            expect(articles.comment_count).toBe(2);
+          }
+          if (articles.article_id === 4) {
+            expect(articles.comment_count).toBe(0);
+          }
         });
-      });
-  });
-  test("GET 200: The articles should be sorted by date in descending order", () => {
-    return request(app)
-      .get("/api/articles")
-      .expect(200)
-      .then(({ body }) => {
-        const { articles } = body;
-        expect(articles).toBeSorted({ key: "created_at", descending: true });
-      });
-  });
-  test("GET 404: Responds with an error if the endpoint is not found", () => {
-    return request(app)
-      .get("/api/article_endpoint")
-      .expect(404)
-      .then(({ body }) => {
-        const { message } = body;
-        expect(message).toBe("Endpoint not found");
-      });
+    });
+    test("GET 200: There should not be a body property present on any of the article objects", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          articles.forEach((article) => {
+            expect(typeof article.body).toBe("undefined");
+          });
+        });
+    });
+    test("GET 200: The articles should be sorted by date in descending order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSorted({ key: "created_at", descending: true });
+        });
+    });
+    test("GET 200: Should accept a topic query and filter articles by the specified topic value", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles.length).toBe(12);
+          articles.forEach((article) => {
+            expect(article.topic).toBe("mitch");
+          });
+        });
+    });
+    test("GET 404: Responds with an error if the endpoint is not found", () => {
+      return request(app)
+        .get("/api/article_endpoint")
+        .expect(404)
+        .then(({ body }) => {
+          const { message } = body;
+          expect(message).toBe("Endpoint not found");
+        });
+    });
+    test("GET 404: Should return a 404 error if a topic does not exist in the database", () => {
+      return request(app)
+        .get("/api/articles?topic=dogs")
+        .expect(404)
+        .then(({ body }) => {
+          const { message } = body;
+          expect(message).toBe("Topic not found");
+        });
+    });
```

Comment:
What if the topic does exist but there are no articles associated with it? (look at 'paper' in the test data!)